src/pgduckdb_background_worker.cpp

```diff
@@ -773,6 +786,26 @@ CreateTable(const char *postgres_schema_name, const char *table_name, const char
 		return false;
 	}
 
+	// Record the dependency on the MotherDuck server
+	{
+		HeapTuple new_table_tuple =
+		    SearchSysCache2(RELNAMENSP, CStringGetDatum(table_name), ObjectIdGetDatum(schema_oid));
+		if (HeapTupleIsValid(new_table_tuple)) {
+			Form_pg_class postgres_relation = (Form_pg_class)GETSTRUCT(new_table_tuple);
+			ReleaseSysCache(new_table_tuple);
+			ObjectAddress object_address = {
+			    .classId = RelationRelationId,
+			    .objectId = GetOid(postgres_relation),
+			    .objectSubId = 0,
+			};
+			recordDependencyOnMDServer(&object_address);
+		} else {
+			// Something went really wrong (since `create_table_succeeded`)
+			elog(WARNING, "Failed to find table '%s.%s' so won't record dependency to 'motherduck' SERVER", table_name,
+			     postgres_schema_name);
```

Comment:
```suggestion
			     postgres_schema_name);
			     RollbackAndReleaseCurrentSubTransaction();
				 return false;
```

And as a nit: then we can change this to "early return" style if we change
```
if (HeapTupleIsValid(new_table_tuple)) {
```
to the inverse
```
if (!HeapTupleIsValid(new_table_tuple)) {